Articles/Blog/2024-01-zfp-sse.md

```diff
@@ -0,0 +1,107 @@
+
+In this sense, I feel like learning on the job can be a good thing.
+On the other hand, I find that we sometimes spend a lot of time searching for answers to deal with oddball continuous integration (CI) and software tool issues.
+It would be nice to have access to a help desk or hotline to get questions like these answered quickly from experts to cut down on time spent searching for answers.
+Sometimes Slack helps to get such questions answered, sometimes not.
+
+**Mark:** My own experience with even the free version of ChatGPT is that it often helps me to find a key entry point into a large body of documentation on a topic I am not familiar with.
+For example, I used it to help guide my adding OpenMP parallelism to the [hello-numerical-world](https://github.com/markcmiller86/hello-numerical-world/tree/main) example and it got me, an OpenMP novice at the time, up and running in less than an hour.
+I think if I had had to read through documentation, it would have been more than 4 hours.
+That said, the free version of ChatGPT doesn't know anything newer than ~2021.
+
+**Peter:** From a personal perspective, I remember feeling a bit overwhelmed having to learn ReadTheDocs, Sphinx, and RST to write zfp documentation from scratch in only a few weeks to meet an ECP deadline while Markus was working on writing tests and hooking up Travis CI.
+In retrospect, it really wasn't that bad--everything seems easy once you know it--but it can seem challenging when you don't even know where to start.
+I was very close to going with tools I was more familiar with (LaTeX, HTML) and alternatives (e.g., Doxygen) but am glad I bit the bullet and went with ReadTheDocs, Sphinx and RST instead.
+With the amount of documentation we now have, it would have been impossible to scale things up with a less capable documentation toolchain.
+
+**Mark:** Humans are often bad at judging cost/benefit trade offs.
+When the [VisIt](https://visit.llnl.gov) project [transitioned from Subversion to git/GitHub](https://bssw.io/blog_posts/continuous-technology-refreshment-an-introduction-using-recent-tech-refresh-experiences-on-visit), we were all worried about the added time and effort needed to review each other's pull requests.
+Within just a week or two of being on GitHub, we had caught enough problems in pull request reviews that we were quickly convinced the cost was worth the benefit.
+
+**Mark:** Where do you think SSE practices on zfp compare with other similar projects you are aware of or familiar with?
+
+**Peter:** When it comes to SSE practices (e.g. testing, documentation, code quality, and portability), I think zfp is in a good place when compared with other, similar packages.
+Developers working on similar packages have even commented that zfp works out of the box without hitches while other packages do not fair so well -- something even documented in published papers.
```

Comment:
fair -> fare

Reply:
Fixed in c9cda8b8a4678cc5efd79a19c8e75c13a58dfccd